modes.xml

```diff
@@ -35,23 +35,31 @@
       <program name="lt-proc -w">
         <file name="spa.automorf.bin"/>
       </program>
-      <program name="cg-proc -w">
+      <program name="cg-proc">
         <file name="spa.rlx.bin"/>
       </program>
-      <program name="apertium-tagger -g">
+      <program name="apertium-tagger -g $2 -p">
         <file name="spa.prob"/>
       </program>
+      <program name="apertium-extract-caps"/>
     </pipeline>
   </mode>
 
-  <mode name="spa-tagger" install="yes">
+  <mode name="spa-tagger" install="yes" gendebug="yes">
     <pipeline>
       <program name="lt-proc -w">
         <file name="spa.automorf.bin"/>
       </program>
-      <program name="apertium-tagger -g">
+      <program name="apertium-tagger -g $2 -p">
         <file name="spa.prob"/>
       </program>
+      <program name="apertium-extract-caps"/>
+      <program name="lt-proc -b">
+        <file name="spa.autogen.bin"/>
+      </program>
```

Comment:
adding a generator step to a tagger pipeline completely changes the semantics of the mode

Reply:
what would be the step between extract and restore -caps?

Reply:
caps handling probably just shouldn't be in this mode - you can add this sequence as a separate mode for testing if you want